assignment3.rb

```diff
@@ -0,0 +1,60 @@
+# ASSIGNMENT 3
+# Create a calculator application which can perform add, sub, division and multiplication operation. Add exception handling for operations like divide which handles ZeroDivisionError exception printing some user friendly message. Add a method in this calculator which takes a number as argument and returns:
+# “positive” - if number is positive
+# “negative” - if number is negative
+# nil - if number is 0
+
+class Calculator
+   def initialize(a,b)
+      @a = a
+      @b = b
+   end
+   def add
+     c = @a + @b
+     "addition of #{@a} and #{@b} is #{c}"
+   end
+   def sub
+     c = @a - @b
+     "substraction of #{@a} and #{@b} is #{c}"
+   end
+   def mul
+     c = @a * @b
+     "multiplication of #{@a} and #{@b} is #{c}"
+   end
+   def division
+     begin
+       c = @a / @b
+       "division of #{@a} and #{@b} is #{c}"
+     rescue ZeroDivisionError => e
+      "#{e.class}: #{e.message}"
+     end
+   end
+  end
+
+class Argument
+  def initialize argument
+    @argument = argument
+  end
+   def check_number
+     if @argument > 0
+       "positive"
+     elsif @argument < 0
+       "negative"
+     elsif @argument == nil
+      nil
+     end
+   end
```

Comment:
@monalis-cuelogic : What will this method return when I call following statement ?

```
arg1 = Argument.new("test")
arg1.check_number
```

Reply:
@monalis-cuelogic : Read the question once again. You should return nil when I pass 0 as argument. You are returning nil when you are getting nil as argument.